src/components/replica/index.tsx

```diff
@@ -129,92 +160,147 @@ const Overview: FunctionComponent<
+    {isCollapsibleLog ? (
       <Accordion className="accordion elevated" chevronPosition="right">
-        <Accordion.Item isExpanded>
+        <Accordion.Item isExpanded={isLogExpanded ?? true}>
           <Accordion.Header>
             <Accordion.HeaderTitle>
               <Typography variant="h4" as="span">
-                Overview
+                Log
               </Typography>
             </Accordion.HeaderTitle>
           </Accordion.Header>
           <Accordion.Panel>
-            <Overview {...rest} />
+            <Code copy autoscroll resizable download downloadCb={downloadCb}>
+              {log || logState?.data}
+            </Code>
           </Accordion.Panel>
         </Accordion.Item>
       </Accordion>
     ) : (
-      <>
-        <Typography variant="h4">Overview</Typography>
-        <Overview {...rest} />
-      </>
+      <Code copy autoscroll resizable download downloadCb={downloadCb}>
+        {log || logState?.data}
+      </Code>
     )}
+  </>
+);
 
-    <section>
-      <AsyncResource asyncState={logState} errorContent={'No log or replica'}>
-        {logState.data ? (
-          isCollapsibleLog ? (
-            <Accordion className="accordion elevated" chevronPosition="right">
-              <Accordion.Item isExpanded>
-                <Accordion.Header>
-                  <Accordion.HeaderTitle>
-                    <Typography variant="h4" as="span">
-                      Log
-                    </Typography>
-                  </Accordion.HeaderTitle>
-                </Accordion.Header>
-                <Accordion.Panel>
-                  <Code
-                    copy
-                    autoscroll
-                    resizable
-                    download
-                    downloadCb={downloadCb}
-                  >
-                    {logState.data}
-                  </Code>
-                </Accordion.Panel>
-              </Accordion.Item>
-            </Accordion>
-          ) : (
-            <Code copy autoscroll resizable download downloadCb={downloadCb}>
-              {logState.data}
-            </Code>
-          )
+export const Replica: FunctionComponent<
+  {
+    header?: string;
+    replica: ReplicaSummary;
+    logState?: FetchQueryResult<string>;
+    getLog?: () => Promise<string>;
+    isCollapsibleOverview?: boolean;
+    isCollapsibleLog?: boolean;
+    downloadCb?: () => void;
+    downloadHistoryCb?: () => void;
+    isLogExpanded?: boolean;
+    getHistoryLog?: () => Promise<unknown>;
+  } & ReplicaElements
+> = ({
+  header,
+  logState,
+  getLog,
+  isCollapsibleOverview,
+  isCollapsibleLog,
+  downloadCb,
+  downloadHistoryCb,
+  isLogExpanded,
+  getHistoryLog,
+  replica,
+  ...rest
+}) => {
+  const [log, setLog] = useState('');
+  const [historyLog, setHistoryLog] = useState('');
+
+  useEffect(() => {
+    if (logState?.data) {
+      return;
+    }
+    getLog?.().then(setLog);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
Replace with `react-hooks/exhaustive-deps` to disable inspection

```suggestion
    // eslint-disable-next-line react-hooks/exhaustive-deps
```